src/hardware/imfc.cpp

```diff
@@ -13376,8 +13376,10 @@ static void imfc_init(Section* sec)
 #endif
 
 	// Register the Audio channel
+	constexpr uint16_t imfc_sampling_rate_hz = 44100;
```

Comment:
So when this is not set to 44.1kHz, the generated pitch does not match that of real-world FB-01 / IMFC recordings?

Having to use 44.1kHz render or "emulation" resolution seems fishy. For one, I could not find any info on the internal sample rate of tge Yamaha PSGs used in the FB-01 / IMFC. I doubt it's 44.1kHz; that is very much tied to CD Audio. But this is an FM sound generator, probably it has some weird internal rate like OPL3.

If you really must set 44.1k, the only explanation I can think of is that the author of the IMFC emulation hardcoded the 44.1k requirement, probably because had a soundcard thst could only do 44.1k or something.

So yeah, I'd like to understand this a bit better.

Reply:
yes; the emulation itself assumes a 44.1 KHz playback sampling rate (or that we "tick over" the emulation at this pace, otherwise the pitch is affected.

There are no obvious ratios / constants / or configurable variables  that set this; so I can only assume this was an assumption i IBM's firmware itself (the emulation was 1:1 reverse engineering of that firmware at the ASM level).

Reply:
Makes sense, maybe a comment would help there to clarify this (saying that the emulation _must_ operate at 44.1kHz).

Reply:
Good call; otherwise it's asking for someone to make it run at whatever the mixer-rate is :-)

Reply:
Added.

Reply:
> Good call; otherwise it's asking for someone to make it run at whatever the mixer-rate is :-)

Yeah, I'd do that next time I touch this file if the comment was not there 😅